File: php/utility/PreparePath.php
<?php
declare(strict_types=1);

// ArcgisHubWorldCountriesGeneralized SDK utility: prepare_path

class ArcgisHubWorldCountriesGeneralizedPreparePath
{
    public static function call(ArcgisHubWorldCountriesGeneralizedContext $ctx): string
    {
        $point = $ctx->point;
        $parts = [];
        if ($point) {
            $p = \Voxgig\Struct\Struct::getprop($point, 'parts');
            if (is_array($p)) {
                $parts = $p;
            }
        }
        return \Voxgig\Struct\Struct::join($parts, '/', true);
    }
}

File: go/utility/prepare_path.go
package utility

import (
	vs "github.com/voxgig-sdk/arcgis-hub-world-countries-generalized-sdk/go/utility/struct"

	"github.com/voxgig-sdk/arcgis-hub-world-countries-generalized-sdk/go/core"
)

func preparePathUtil(ctx *core.Context) string {
	point := ctx.Point

	var parts []any
	if p := vs.GetProp(point, "parts"); p != nil {
		if pl, ok := p.([]any); ok {
			parts = pl
		}
	}

	path := vs.Join(parts, "/", true)
	return path
}
